### PR TITLE
Development

### PR DESCRIPTION
**Pull request by tomishay**

### CODE REVIEW

**Comment by tomishay** [comment]
[This issue ](https://github.com/jasp-stats/jaspProcessControl/issues/162) lists the changes made to each analysis in the commits.

File: R/doeResponseSurfaceMethodology.R [hunk 1]
```diff
@@ -317,51 +315,56 @@ doeResponseSurfaceMethodology <- function(jaspResults, dataset, options, ...){
     }
   }
 
-
+  set.seed(1)
```
Comment (vandenman) [inline R/doeResponseSurfaceMethodology.R:318]: is this intended? You can also add an option for a seed in the GUI if you want to ensure reproducibility (there is a QML component for this).

File: R/doeFactorial.R [hunk 2]
```diff
@@ -267,15 +278,30 @@ doeFactorial <- function(jaspResults, dataset, options, ...){
...
+      else if (any(sumResolution == resolution.Five))
+        res <- "V"
+      else if (any(sumResolution == resolution.three))
+        res <- "III"
+      else if (any(sumResolution == 137))
+        res <- "VIII"
+      else if (any(sumResolution == 71))
+        res <- "VII"
+      else
+        res <- "III"
```
Comment (vandenman) [inline R/doeFactorial.R:303]: if `sumResolution` is a single number, this should be
```r
       if (sumResolution %in% resolution.Full)) 
```
same for the stuff below. If `sumResolution` is not a single number and the order matters, then what you did is fine.

File: R/doeFactorial.R [hunk 3]
```diff
@@ -475,7 +487,12 @@ doeFactorial <- function(jaspResults, dataset, options, ...){
     }
     if(options[["dataCoding"]] == "dataUncoded"){
       for(i in 1:options[["numberOfFactors"]]){
-        rows[,i+2][rows[,i+2] == -1] <- factorLows[i]
+
+        if(any(factorLows[i] == "1"))
+          rows[,i+2][rows[,i+2] == -1] <- "1.0"
+        else
+          rows[,i+2][rows[,i+2] == -1] <- factorLows[i]
```
Comment (vandenman) [inline R/doeFactorial.R:494]:
```suggestion
        rows[,i+2][rows[,i+2] == -1] <- if (any(factorLows[i] == "1"))
          "1.0"
        else
          factorLows[i]
```

File: R/doeFactorial.R [hunk 4]
```diff
@@ -497,6 +514,7 @@ doeFactorial <- function(jaspResults, dataset, options, ...){
     }
 
     if(options[["runOrder"]] == "runOrderRandom"){
+      set.seed(1)
```
Comment (vandenman) [inline R/doeFactorial.R:517]: same thing about the seed.

File: R/doeFactorial.R [hunk 5]
```diff
@@ -842,6 +862,11 @@ doeFactorial <- function(jaspResults, dataset, options, ...){
     )
   }
 
+  # Replace : by \u00d7
+  for (i in 1:length(anovaFill$Terms)) {
+    anovaFill$Terms[i] <- gsub(":", " \u00d7 ", anovaFill$Terms[i])
+  }
```
Comment (vandenman) [inline R/doeFactorial.R:868]: is this for an interaction symbol? if yes, you can replace this by
```suggestion
  anovaFill$Terms <- jaspBase::gsubInteractionSymbol(anovaFill$Terms)
```
which will transform e.g., `"factor:covariate"` into `"factor ✻ covariate"`

File: R/doeFactorial.R [hunk 6]
```diff
@@ -873,9 +898,14 @@ doeFactorial <- function(jaspResults, dataset, options, ...){
...
-    formula <- sprintf("%s = %.5g%s %s %.5g%s", reponseVar, coefs[1], factors[1],plusOrMin[2], abs(coefs[2]), factors[2])
+    formula <- sprintf("%s = %.5g %s %s %.5g %s", reponseVar, coefs[1], factors[1],plusOrMin[2], abs(coefs[2]), factors[2])
     for (i in 3:length(coefs)) {
-      formula <- sprintf("%s %s %.5g%s", formula, plusOrMin[i], abs(coefs[i]), factors[i])
+      formula <- sprintf("%s %s %.5g %s", formula, plusOrMin[i], abs(coefs[i]), factors[i])
+    }
+
+    # Replace : by \u00d7
+    for (i in 1:length(formula)) {
+      formula[i] <- gsub(":", " \u00d7 ", formula[i])
```
Comment (vandenman) [inline R/doeFactorial.R:908]: same stuff about interaction symbols

File: R/doeFactorial.R [hunk 7]
```diff
@@ -919,10 +960,16 @@ doeFactorial <- function(jaspResults, dataset, options, ...){
...
-      p     = rep("*", length(names))
+      p     = rep("*", length(names)),
+      vif     = rep("*", length(names))
     )
   }
 
+  # Replace : by \u00d7
+  for (i in 1:length(coefsFill$terms)) {
+    coefsFill$terms[i] <- gsub(":", " \u00d7 ", coefsFill$terms[i])
+  }
```
Comment (vandenman) [inline R/doeFactorial.R:971]: same about the interaction stuff

**Review by vandenman — CHANGES_REQUESTED** [review]
Looks fine, some minor stuff below.

File: R/doeResponseSurfaceMethodology.R (hunk 1 above)
Comment (tomishay) [reply R/doeResponseSurfaceMethodology.R:318]: Yes, it is intended, I want the random order to always be the same random order, what's the GUI component for it?

Comment (JorisGoosen) [reply R/doeResponseSurfaceMethodology.R:318]: Its called `SetSeed`

**Pull request merged**